### PR TITLE
Update SolverBenchmark for DataFrames v.0.21

### PR DESCRIPTION
**Pull request by amontoison**
supersedes #55

### CODE REVIEW

**Comment by abelsiqueira** [comment]
The "^0.20" means `[0.20.0, 0.21.0)` and "=0.20" means only "0.20.0" (ref: https://julialang.github.io/Pkg.jl/v1/compatibility/). The current "^0.20" is already and upper bound.

**Comment by amontoison** [comment]
Thanks Abel!
The problem is that the last release was done without #37. The Project.toml doesn't need to be updated.

**Comment by abelsiqueira** [comment]
Version 0.2.0 should have Project.toml with upper bounds.

**Comment by abelsiqueira** [comment]
https://github.com/JuliaSmoothOptimizers/SolverBenchmark.jl/releases/tag/v0.2.0

**Comment by amontoison** [comment]
Hum, I don't understand why I have DataFrames v0.21 installed.

**Comment by amontoison** [comment]
Version 0.2.0 should have Project.toml with upper bounds but not version 0.1.0 :laughing: 
solverbenchmarks

**Comment by abelsiqueira** [comment]
I guess it has to decide which package will be downgraded, and it chose SolverBenchmark. Not much we can do except update to handle `DataFrames 0.21`.

**Comment by amontoison** [comment]
`flag` became `status` in some tests, is it a modification of DataFrames package or I forgot to update something here?

File: src/join.jl [hunk 1]
```diff
@@ -49,13 +49,14 @@ function join(stats::Dict{Symbol,DataFrame},
   df = stats[s][:, invariant_cols]
 
   rename_f(c, s) = begin
-    c in invariant_cols && return c
-    sc = haskey(hdr_override, c) ? hdr_override[c] : string(c)
+    Symbol(c) in invariant_cols && return c
+    sc = haskey(hdr_override, c) ? hdr_override[c] : c
```
Comment (amontoison) [inline src/join.jl:53]: `haskey(hdr_override, Symbol(c))`

**Comment by amontoison** [comment]
I don't understand why PkgBenchmark says that the package is dirty in the tests...

File: Project.toml [hunk 2]
```diff
@@ -16,11 +16,12 @@ Plots = "91a5bcdd-55d7-5caf-9e0b-520d859cae80"
 PrettyTables = "08abe8d2-0d0c-5749-adfa-8a2ac140af0d"
 Printf = "de0858da-6303-5e67-8744-51eddeeeb8d7"
 Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
+SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
```
Comment (dpo) [inline Project.toml:19]: Why do we need SparseArrays and we didn't need it before?

**Comment by dpo** [comment]
> I don't understand why PkgBenchmark says that the package is dirty in the tests...

Try adding `benchmark/*.json` to `.gitignore`. I'm not sure why it wasn't an issue before.

File: Project.toml (hunk 2 above)
Comment (amontoison) [reply Project.toml:19]: I suppose that they fixed some type piracy with the release 1.4.1 : https://github.com/JuliaLang/julia/issues/30945 or https://github.com/JuliaLang/julia/issues/35899.

**Comment by dpo** [comment]
Seems like PkgBenchmark is leaving more stuff around. Add a `git status` to `.travis.yml` to debug.

**Comment by amontoison** [comment]
It should work now. When the package was cloned, the Project.toml was regenerated differently...

**Comment by abelsiqueira** [comment]
Solving manually.

**Comment by abelsiqueira** [comment]
Merged manually: f57c8028dc72472236a10603a49d9bc6a6c7b381
Thanks, @amontoison

**Pull request closed without merging**